### PR TITLE
[TEVA-3361] AB test alternative search results

### PR DESCRIPTION
**Pull request by robyoung91**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-3361

## Changes in this PR:

- This PR adds an AB test variant which moves the search fields from the left column to the right column on the results page _for desktop users only_

### Before
image

### After
image

### CODE REVIEW

File: app/components/panel_component/panel.scss [hunk 1]
```diff
@@ -28,6 +28,16 @@
       }
     }
   }
+
+  .search-controls__alternative {
```
Comment (david-mears-dfe) [inline app/components/panel_component/panel.scss:32]: I think we should use double hyphen here. We're following this convention: https://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/

File: app/views/subscriptions/_fields.html.slim [hunk 2]
```diff
@@ -1,5 +1,5 @@
 = render "shared/search/keyword", f: f, wide: false
-= render "shared/search/location", f: f, wide: false
+= render "shared/search/location", f: f, hint: true, wide: false
```
Comment (david-mears-dfe) [inline app/views/subscriptions/_fields.html.slim:2]: Looks like `wide` doesn't do anything so we can get rid.

File: config/ab_tests.yml [hunk 3]
```diff
@@ -50,3 +53,6 @@ test:
   2021_10_mobile_search_panel_test:
     present: 1
     overlay: 0
+  2021_11_desktop_search_results_page_test:
+    present: 1
+    alternative: 0
```
Comment (david-mears-dfe) [inline config/ab_tests.yml:58]: Doesn't this mean 'always use the present version and never the alternative'?

Comment (robyoung91) [reply config/ab_tests.yml:58]: Yes - I believe we set it like that under `test:` so we don't get the variants complicating things when we are running tests?

File: app/components/panel_component/panel.scss (hunk 1 above)
Comment (alexbowen) [reply app/components/panel_component/panel.scss]: yes correct @david-mears-dfe - alternative is a modifier strictly speaking

File: Dockerfile [hunk 4]
```diff
@@ -1,11 +1,11 @@
-ARG PROD_PACKAGES="libxml2=2.9.12-r1 libxslt=1.1.34-r1 libpq=13.4-r0 tzdata=2021e-r0 shared-mime-info=2.1-r0 util-linux=2.37.2-r0 busybox=1.33.1-r6"
```
Comment (alexbowen) [inline Dockerfile:1]: this file change should not be in this PR, some a bit screwy happened withe your rebasing

File: app/frontend/src/styles/application/jobseekers/search-header.scss [hunk 5]
```diff
@@ -31,3 +50,9 @@
     }
   }
 }
+
+div.grid-column-right .search-results__header {
```
Comment (alexbowen) [inline app/frontend/src/styles/application/jobseekers/search-header.scss:54]: can we remove the `div.` from this selector, i do not think it is neccessary

File: config/ab_tests.yml (hunk 3 above)
Comment (alexbowen) [inline config/ab_tests.yml:58]: i would rather these names were a bit more descriptive `alternative` doesnt mean much to me, whereas `right` or `right_column` for example gives a bit more clue to whats going on

**Comment by alexbowen** [comment]
doesnt appear to be working on ipad safari, probably needs a media query tweak - sorry
Screenshot 2021-11-16 at 21 39 54

**Comment by alexbowen** [comment]
have tested in samsung android, iphone safari and ipad safari on browserstack - clean bill of health

**Review by cesidio — APPROVED** [review]

**Comment by alexbowen** [comment]
this is good with me but for someone else can approve as i worked on it

**Pull request merged**